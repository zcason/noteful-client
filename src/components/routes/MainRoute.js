import React, { Component } from 'react';
import MainSection from '../sections/MainSection';
import SidebarSection from '../sections/SidebarSection';
import { withRouter } from 'react-router-dom';
import ErrorBoundary from '../../errors/ErrorBoundary';
import PropTypes from 'prop-types';

class MainRoute extends Component {
    render(){
        return (
            <div className="App">
                <ErrorBoundary message="Sidebar Section Error">
                    <SidebarSection 
                        store={this.props.store}
                    />
                </ErrorBoundary>
                <ErrorBoundary message="Main Section Error">
                    <MainSection 
                        store={this.props.store}
                    />
                </ErrorBoundary>
            </div>
       );
    }
}

MainRoute.propTypes = {
    store: PropTypes.object.isRequired
}

export default withRouter(MainRoute);