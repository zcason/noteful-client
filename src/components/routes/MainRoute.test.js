import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../../dummy-store';
import MainRoute from './MainRoute';

describe(`MainRoute component`, () => {
    const props = {
      store: store
    }
  
    it('renders MainRoute by default', () => {
      const wrapper = shallow(<MainRoute />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders main route from store when in props', () => {
      const component = shallow(<MainRoute {...props} />)
        .find('.App')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  