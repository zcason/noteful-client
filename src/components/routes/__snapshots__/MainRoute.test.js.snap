// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainRoute component renders MainRoute by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`MainRoute component renders main route from store when in props 1`] = `null`;
